src/input/input_json.rs

```diff
@@ -399,30 +380,34 @@ impl<'a> Input<'a> for String {
         JsonValue::parse(self.as_bytes(), true).map_err(|e| map_json_err(self, e))
     }
 
-    fn strict_str(&'a self) -> ValResult<EitherString<'a>> {
-        Ok(self.as_str().into())
+    fn validate_str(
+        &'a self,
+        _strict: bool,
+        _coerce_numbers_to_str: bool,
+    ) -> ValResult<ValidationMatch<EitherString<'a>>> {
+        // Justification for `strict` instead of `exact` is that in JSON strings can also
+        // represent other datatypes such as UUID and date more exactly, so string is a
+        // converting input
+        Ok(ValidationMatch::strict(self.as_str().into()))
     }
 
     fn strict_bytes(&'a self) -> ValResult<EitherBytes<'a>> {
         Ok(self.as_bytes().into())
     }
 
-    fn strict_bool(&self) -> ValResult<bool> {
-        str_as_bool(self, self)
+    fn validate_bool(&self, _strict: bool) -> ValResult<'_, ValidationMatch<bool>> {
+        str_as_bool(self, self).map(ValidationMatch::strict)
     }
 
-    fn strict_int(&'a self) -> ValResult<EitherInt<'a>> {
+    fn validate_int(&'a self, _strict: bool) -> ValResult<'a, ValidationMatch<EitherInt<'a>>> {
         match self.parse() {
-            Ok(i) => Ok(EitherInt::I64(i)),
+            Ok(i) => Ok(ValidationMatch::strict(EitherInt::I64(i))),
```

Comment:
same?

Reply:
Ditto as above, this came from `strict_int`.